File: src/main/java/me/pcasaes/hexoids/core/domain/index/PlayerSpatialIndexFactory.java
package me.pcasaes.hexoids.core.domain.index;

public class PlayerSpatialIndexFactory {

    private static final PlayerSpatialIndexFactory FACTORY = new PlayerSpatialIndexFactory();

    private PlayerSpatialIndexFactory() {
    }

    public static PlayerSpatialIndexFactory factory() {
        return FACTORY;
    }

    private PlayerSpatialIndex playerSpatialIndex;

    public PlayerSpatialIndex get() {
        return playerSpatialIndex;
    }

    public void setPlayerSpatialIndex(PlayerSpatialIndex playerSpatialIndex) {
        this.playerSpatialIndex = playerSpatialIndex;
    }
}

File: src/main/kotlin/me/pcasaes/hexoids/core/domain/index/PlayerSpatialIndexFactory.kt
package me.pcasaes.hexoids.core.domain.index

object PlayerSpatialIndexFactory {

    private lateinit var playerSpatialIndex: PlayerSpatialIndex

    fun get(): PlayerSpatialIndex {
        return playerSpatialIndex
    }

    fun setPlayerSpatialIndex(playerSpatialIndex: PlayerSpatialIndex) {
        this.playerSpatialIndex = playerSpatialIndex
    }
}
